src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -382,11 +382,30 @@ public static T[] AsArray<T>(this IEnumerable<T> source)
             return builder.ToImmutableAndFree();
         }
 
+        public static ImmutableArray<TResult> SelectManyAsArray<TSource, TResult>(this IEnumerable<TSource>? source, Func<TSource, IEnumerable<TResult>> selector)
```

Comment:
>SelectManyAsArray [](http://example.com/codeflow?start=46&length=17)

A version of this that takes TArg and passes it to the selector would be useful.

Reply:
sure.  i'll add.

Reply:
done.